src/sql/base/browser/ui/radioButton/radioButton.ts

```diff
@@ -33,7 +36,23 @@ export class RadioButton extends Widget {
 		this.label = opts.label;
 		this.enabled = opts.enabled || true;
 		this.checked = opts.checked || false;
-		this.onclick(this.inputElement, () => this._onClicked.fire());
+		this.onclick(this.inputElement, () => {
+			this._onClicked.fire();
+			if (this.name) {
+				let buttonGroup = document.getElementsByName(this.name);
+				buttonGroup.forEach((button) => {
+					let event = document.createEvent('HTMLEvents');
+					event.initEvent('change', true, true);
+					button.dispatchEvent(event);
+				});
+			}
+		});
+		this.inputElement.addEventListener('change', () => {
+			if (this.inputElement.checked !== this._internalCheckedStateTracker) {
+				this.checked = this.inputElement.checked;
+				this._onChanged.fire(this.checked);
```

Comment:
This should probably go in the checked setter  so it's correctly fired when something sets that directly. (I'd also move the internal state logic there too)

Reply:
Should I fire the event even when some external code updates the 'checked' variable?
I was trying to avoid that scenario

Reply:
Why wouldn't you? onChanged should be fired any time the selected state is changed - that's the whole point of this code.